File: 백준/Silver/14732. 행사장 대여 （Small）/행사장 대여 （Small）.py
import sys
input = sys.stdin.readline

n = int(input())
visited = [[0 for _ in range(501)] for _ in range(501)]

for j in range(n):
    x1, y1, x2, y2 = map(int, input().split())

    for i in range(x1, x2):
        for j in range(y1, y2):
            visited[i][j] = 1

res = 0
for i in range(501):
    for j in range(501):
        if visited[i][j] == 1:
            res += 1

print(res)

File: 백준/Silver/14732. 행사장 대여 （Small）/행사장 대여 （Small）.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./example.txt")
  .toString()
  .trim()
  .split("\n")
  .map((v) => v.split(" ").map(Number));

const [n, ...mp] = input;
const map = Array.from(Array(501), () => Array(501).fill(0));

for (const v of mp) {
  let [x1, y1, x2, y2] = v;
  for (let i = x1; i < x2; i++) {
    for (let j = y1; j < y2; j++) {
      map[i][j] = 1;
    }
  }
}

let res = 0;
for (let i = 0; i <= 500; i++) {
  for (let j = 0; j <= 500; j++) {
    if (map[i][j]) res += 1;
  }
}
console.log(res);
